Library/Homebrew/utils/github.rb

```diff
@@ -50,24 +50,34 @@ def api_credentials
       if ENV["HOMEBREW_GITHUB_API_TOKEN"]
         ENV["HOMEBREW_GITHUB_API_TOKEN"]
       elsif ENV["HOMEBREW_GITHUB_API_USERNAME"] && ENV["HOMEBREW_GITHUB_API_PASSWORD"]
-        [ENV["HOMEBREW_GITHUB_API_USERNAME"], ENV["HOMEBREW_GITHUB_API_PASSWORD"]]
+        [ENV["HOMEBREW_GITHUB_API_PASSWORD"], ENV["HOMEBREW_GITHUB_API_USERNAME"]]
       else
-        github_credentials = Utils.popen("git credential-osxkeychain get", "w+") do |io|
-          io.puts "protocol=https\nhost=github.com"
-          io.close_write
-          io.read
-        end
-        github_username = github_credentials[/username=(.+)/, 1]
-        github_password = github_credentials[/password=(.+)/, 1]
-        if github_username && github_password
-          [github_password, github_username]
-        else
-          []
-        end
+        api_credentials_from_keychain("https", "github.com")
       end
     end
   end
 
+  def api_credentials_from_keychain(protocol, host)
+    credentials = Utils.popen("git credential-osxkeychain get", "w+") do |pipe|
+      pipe.write "protocol=#{protocol}\nhost=#{host}\n"
+      pipe.close_write
+      pipe.read
+    end
+
+    credentials = Hash[credentials.scan(/^(username|password)=(.*)$/)]
```

Comment:
I think the `Hash[]` approach is less obvious in what it's actually doing, personally.